tasks/symlinks.yml

```diff
@@ -0,0 +1,37 @@
+- name: get all subdirectories
+  find:
+    paths: "{{service_item}}/live"
+    recurse: no
+    file_type: directory
+  register: service_subdirs
+  become: yes
+  become_user: root
+
+- name: get all hosts
+  set_fact:
+    service_hosts: "{{service_subdirs.files | map(attribute='path') | map('regex_replace', service_item + '/live/') | list}}"
+
+- name: "stat a fullchain_privkey file in {{service_item}}"
+  stat:
+    path: "{{service_item}}/live/{{item}}/fullchain_privkey.pem"
+  register: src_files
+  loop: "{{service_hosts}}"
+  become: yes
+  become_user: root
+
```

Comment:
I wouldn't bother checking. Just assume it is there and make the symlink. A broken link should not be a problem.

Reply:
If `update.py` (auto-cert) or letsencrypt's service failed to create the file, wouldn't I want to know instead of having a silent failure?

Reply:
No. It is up to lets encrypt and and auto-cert to make sure the are doing what they are supposed to. You are not testing the lets encrypt and auto-cert roles in the haproxy role. There are many other failure modes, e.g. not creating the live directory or corrupt certificates, what makes this specific one special?